### PR TITLE
BlogService: Refactor for syncing a blog, settings, and options

### LINKED ISSUE #6120: Site Settings should be automatically synced
Currently, `blog.settings` is only synced when a user visits the "Settings" for their site within the app. Without this sync, we don't have the site's default category, post format, or other such settings.

Example: https://github.com/wordpress-mobile/WordPress-iOS/pull/6114

We should find a timely opportunity within the app's lifecycle to go ahead and a sync a site's settings, and frequently sync to keep it up to date. We should also not rely on syncing settings only when they're needed, such as when a user creates a post, but instead _before_ they even try to create a post.

Potential sync timing:
- Sync settings on initial log in or when a site is added.
- Sync settings on every app launch.
- Sync settings on every app foregrounding, with a minimum timing frequency.
- Sync settings when a user switches sites on the My Sites tab.

### PR DESCRIPTION
**Pull request by kurzee**
Fixes #6120 

This refactor largely started to address the issue seen in #6120. However while working through the issue and where a sync can occur, it was quickly apparent we have a decent size of tech debt surrounding the different methods of syncing a blog, syncing it's settings, or options or details. I've run into this area of confusion before but never within the right context to take a moment and refactor.

The bulk of this PR moves around code, renames methods and updates code documentation comments. The primary goal was to better separate the concerns and paths between the `BlogServiceRemoteREST` and the `BlogServiceRemoteXMLRPC` and which remote can handle which type of sync or update.

Particularly, methods were moved out of the top-level `BlogServiceRemote` protocol and more clearly included/written into the respective remote classes. For example, the XMLRPC remote handles the bulks of it's syncing through the `options` fields on the site while the REST remote handles syncing through a site endpoint itself, or a settings endpoint. The separation and translation here is now handled within `BlogService` to have better type-safety and clear implementation requirements.

Lastly, to finally resolved the issue in #6120, we run a settings sync on the "sync all" method now named as `syncBlogAndAllMetadata` to ensure that a site's settings are synced regularly. Particularly, this allows us to know the default category and post format without ever visiting the settings controller.

To test:
1. Uninstall the app.
2. Add a self-hosted site.
3. Ensure the editor options has all categories and post formats available for selection.
4. Go to the Blog Details > Settings view.
5. Ensure the settings are as expected from the site.
6. Update the Site name and tagline, check that the web reflects the update made in the app.
7. Quit the app.
8. Change the site name and tagline on the web.
9. Launch the app, check that the name and tagline are updated once the Blog Details is visited and also the Settings view.

To test:
1. Uninstall the app.
2. Log into a .com site.
3. Open the editor, then the options.
4. Ensure that the default category (blank if uncategorized) is set, as well as the post format.
5. Open the Blog Details > Settings view.
6. Ensure that the default category (or uncategorized) is set, as well as the post format.
7. Change the default category and post format.
8. Check that the change is reflected on the web.
9. Quite the app.
10. On the web, change the default category and post format.
11. Launch the app.
12. Open the editor, then the options.
13. Ensure that the default category you selected on the web is set, as well as the post format.
14. Open the Blog Details > Settings view.
15. Ensure that the default category is set, as well as the post format.

Needs review: @frosty can you take a stroll through this refactor and fix? My apologies that this and the diff may look a bit crazy. A fair amount of the code was simply moved for ideal positioning relative to the logic of the method.

### CODE REVIEW

**Comment by frosty** [comment]
Hey @kurzee! Still walking through tests and the code, but I've tested the self-hosted steps you gave and I ran into a few issues:

* After logging in, I opened the editor and went to post settings. Post formats showed 'Unavailable', and the categories list was empty.
* They still don't seem to appear after going into Settings, navigating out of and back into Blog Details, or backgrounding the app... the only way I could get them to show up was quitting and reopening the app.
* When I updated the site name and tagline on the web, the title of the Blog Details screen updated as expected, but the title show in the blog details header view did not update.

**Comment by frosty** [comment]
Seems to be working just fine for dotcom, though! 👍

File: WordPress/Classes/Services/BlogService.h [hunk 1]
```diff
@@ -50,18 +50,57 @@ extern NSString *const WordPressMinimumVersion;
  */
 - (nullable Blog *)primaryBlog;
 
+/**
+ *  Sync all available blogs for an acccount
+ *
+ *  @param account the account for the associated blogs.
+ *  @param success a block that is invoked when the sync is sucessfull.
```
Comment (frosty) [inline WordPress/Classes/Services/BlogService.h]: ❤️ all these doc comments! But you need one extra 'c' and one less 'l' in every 'successful'.

**Comment by kurzee** [comment]
>When I updated the site name and tagline on the web, the title of the Blog Details screen updated as expected, but the title show in the blog details header view did not update.

@frosty nice catch. I went ahead and snuck in a fix for this, we weren't actually looking for changes to `blog.settings` in the `BlogDetailsViewController` so it never refreshed properly.

**Comment by kurzee** [comment]
For posterity, I realized and remembered that you can't actually _get_ the default post category or formats via XML-RPC. Striking out the tests above.

**Comment by kurzee** [comment]
Merging as per approval from @frosty and passing tests 😃 

Thanks @frosty!

**Pull request merged**